specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,294 @@
+            CHECK_FAILURE(args->get_Request(&webResourceRequest));
+            // The response object as received
+            wil::com_ptr<ICoreWebView2WebResourceResponseView> webResourceResponse;
+            CHECK_FAILURE(args->get_Response(&webResourceResponse));
+            
+            // Get body content for the response. Redirect responses will
+            // return an error HRESULT as their body (if any) is ignored.
+            HRESULT getContentCallResult = webResourceResponse->TryGetContent(
+                Callback<
+                    ICoreWebView2WebResourceResponseViewTryGetContentCompletedHandler>(
+                    [this, webResourceRequest, webResourceResponse](HRESULT result, IStream* content) {
+                        // The response might not have content, or it might have failed loading.
+                        bool gotContent = SUCCEEDED(result);
+                        
+                        std::wstring message =
+                            L"{ \"kind\": \"event\", \"name\": "
+                            L"\"WebResourceResponseReceived\", \"args\": {"
+                            L"\"request\": " +
+                            RequestToJsonString(webResourceRequest.get()) +
+                            L", "
+                            L"\"response\": " +
+                            ResponseToJsonString(webResourceResponse.get(), content) + L"}";
+
+                        message +=
+                            WebViewPropertiesToJsonString(m_webview.get());
+                        message += L"}";
+                        PostEventMessage(message);
+                        return S_OK;
+                    })
+                    .Get());
+
+            return S_OK;
+        })
+        .Get(),
+    &m_webResourceResponseReceivedToken);
+```
+
+## C#
+```c#
+WebView.WebResourceResponseReceived += WebView_WebResourceResponseReceived;
+
+// Note: modifications made to request are set but have no effect on WebView processing it.
+private async void WebView_WebResourceResponseReceived(CoreWebView2 sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
+{
+    // Actual headers sent with request
+    foreach (var current in e.Request.Headers)
+    {
+        Console.WriteLine(current);
+    }
+
+    // Headers in response received
+    foreach (var current in e.Response.Headers)
+    {
+        Console.WriteLine(current);
+    }
+
+    // Status code from response received
+    int status = e.Response.StatusCode;
+    if (status == 200)
+    {
+        // Handle
+        Console.WriteLine("Request succeeded!");
+
+        // Get response body
+        try
+        {
+            System.IO.Stream content = await e.Response.TryGetContentAsync();
+            DoSomethingWithResponseContent(content);
+        }
+        catch (COMException ex)
+        {
+            // A COMException will be thrown if no content was found for the
+            // response, or if it failed to load.
```

Comment:
The `Try` prefix means that it returns `null` on failure, rather than throwing an exception on failure. Is there a way for an app to know ahead of time whether the response contains content?

Reply:
They could try looking for the `Content-Length` header for responses to requests with methods other than `HEAD`.

Reply:
This was updated to use `Get` prefix instead of `TryGet`. It will still throw an exception if the content fails to load.